sound/soc/sof/topology.c

```diff
@@ -1844,7 +1844,8 @@ static int sof_process_load(struct snd_soc_component *scomp, int index,
 			dev_err(sdev->dev, "error: unknown kcontrol type %d in widget %s\n",
 				widget->dobj.widget.kcontrol_type,
 				widget->name);
-			return -EINVAL;
+			ret = -EINVAL;
+			goto err;
```

Comment:
is this a real issue or a false-positive reported by a tool?

In this function, we loop twice on the same control components, and deal with the same switch case, so if there is an unknown kcontrol_type it'd be caught during the first iteration *before* allocating the memory, and we would never hit the second case.

Care to provide more details?

Reply:
Thanks @plbossart 
This issue is reported by a tool.
You're right. Logically, it's impossible for this case to happen. I will mark it as 'not a problem' in the tool and close this PR.

Reply:
> is this a real issue or a false-positive reported by a tool?
> 
> In this function, we loop twice on the same control components, and deal with the same switch case, so if there is an unknown kcontrol_type it'd be caught during the first iteration _before_ allocating the memory, and we would never hit the second case.

This is good point, we won't hit the default case in the 2nd iteration, maybe we should remove this part from the 2nd iteration:
```
-		default:
-			dev_err(sdev->dev, "error: unknown kcontrol type %d in widget %s\n",
-				widget->dobj.widget.kcontrol_type,
-				widget->name);
-			return -EINVAL;
```




> 
> Care to provide more details?

Reply:
Agree with @keyonjie , but may need a comment to explain why we don't need the default case at the 2nd iteration.

Reply:
Thanks @keyonjie  and @bardliao 
Good suggestion, let's just remove the default case.